🛠 Tooling: Missing tsc script in CI

### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/joshuakgoldberg-dot-com-next/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

This repo lints, formats, etc. in CI but doesn't run `tsc`! So right now on `main` there are type checking errors:

```plaintext
$ pnpm tsc

> joshuakgoldberg-dot-com-next@0.0.1 tsc /Users/josh/repos/dot-com
> tsc

src/components/ContentEntry.tsx:2:15 - error TS1484: 'JSX' is a type and must be imported using a type-only import when 'verbatimModuleSyntax' is enabled.

2 import { For, JSX } from "solid-js";
                ~~~

src/components/GroupedEntries.tsx:1:15 - error TS1484: 'JSX' is a type and must be imported using a type-only import when 'verbatimModuleSyntax' is enabled.

1 import { For, JSX } from "solid-js";
                ~~~

src/components/Text.tsx:2:10 - error TS1484: 'JSX' is a type and must be imported using a type-only import when 'verbatimModuleSyntax' is enabled.

2 import { JSX, splitProps } from "solid-js";
           ~~~


Found 3 errors in 3 files.

Errors  Files
     1  src/components/ContentEntry.tsx:2
     1  src/components/GroupedEntries.tsx:1
     1  src/components/Text.tsx:2
```

### Additional Info

_No response_

--- a/.github/workflows/tsc.yml
+++ b/.github/workflows/tsc.yml
@@ -1,0 +1,17 @@
+jobs:
+  tsc:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./.github/actions/prepare
+      - run: pnpm astro sync
+      - run: pnpm tsc
+
+name: Type Check
+
+on:
+  pull_request: ~
+
+  push:
+    branches:
+      - main

--- a/src/components/ContentEntry.tsx
+++ b/src/components/ContentEntry.tsx
@@ -1,5 +1,5 @@
 import { clsx } from "clsx";
-import { For, JSX } from "solid-js";
+import { For, type JSX } from "solid-js";
 
 import { Anchor } from "./Anchor";
 import styles from "./ContentEntry.module.css";

--- a/src/components/GroupedEntries.tsx
+++ b/src/components/GroupedEntries.tsx
@@ -1,4 +1,4 @@
-import { For, JSX } from "solid-js";
+import { For, type JSX } from "solid-js";
 
 import { EntryList } from "./EntryList";
 

--- a/src/components/Text.tsx
+++ b/src/components/Text.tsx
@@ -1,5 +1,5 @@
 import { clsx } from "clsx";
-import { JSX, splitProps } from "solid-js";
+import { type JSX, splitProps } from "solid-js";
 import { Dynamic } from "solid-js/web";
 
 import styles from "./Text.module.css";

